Mathlib/Topology/Separation/DisjointCover.lean

```diff
@@ -0,0 +1,74 @@
+/-
+Copyright (c) 2024 David Loeffler. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: David Loeffler
+-/
+import Mathlib.Order.CompletePartialOrder
+import Mathlib.Order.Disjointed
+import Mathlib.Topology.Separation.Profinite
+import Mathlib.Topology.Sets.Closeds
+
+/-!
+# Disjoint covers of profinite spaces
+
+We show that if `X` is a profinite space, then any open covering of `X` can be refined to a finite
+covering by disjoint clopens.
+-/
+
+open Set
+
+open TopologicalSpace (Opens Clopens)
+
+open scoped Function Finset
+
+variable {ι X : Type*}
+  [TopologicalSpace X] [TotallyDisconnectedSpace X] [T2Space X] [CompactSpace X]
+
+/-- Any open cover of a profinite space can be refined to a finite cover by clopens. -/
+lemma CompactSpace.exists_finite_clopen_cover_of_open_cover
+    {U : ι → Opens X} (hU : univ ⊆ ⋃ i, (U i : Set X)) :
+    ∃ (n : ℕ) (V : Fin n → Clopens X),
+    (∀ j, ∃ i, (V j : Set X) ⊆ U i) ∧ univ ⊆ ⋃ j, (V j : Set X) := by
```

Comment:
Do we have/would it be worth defining what a cover is so that the statements don't need all the ands? I see `HasSeparatingCover` but no cover (unless I'm looking in the wrong place!). Although maybe this would need to be its own PR if its lots of work.

Reply:
I don't think we have a definition of "cover" for topological spaces yet (unless there is something buried in the category / site theoretic stuff?). I fear it might be a bit of a can of worms; lots of people will have firmly held and mutually incompatible views as to what the formulation should be.

Reply:
I've opened a conversation about this on Zulip [here](https://leanprover.zulipchat.com/#narrow/channel/287929-mathlib4/topic/Definition.20of.20a.20covering.20of.20a.20topological.20space.3F).

Reply:
(marking this convo as resolved since #21442 is now merged and this PR rebased on top of it)